src/System.Text.Json/src/System/Text/Json/Node/JsonNumber.cs

```diff
@@ -5,91 +5,161 @@
+        public short GetInt16() => short.Parse(_value);
+        
+        public int GetInt32() => int.Parse(_value);
+        
+        public long GetInt64() => long.Parse(_value);
+        
+        public float GetSingle() => float.Parse(_value);
+        
+        public double GetDouble() => double.Parse(_value);
+        
+        [CLSCompliant(false)]
+        public sbyte GetSByte() => sbyte.Parse(_value);
+        
+        [CLSCompliant(false)]
+        public ushort GetUInt16() => ushort.Parse(_value);
+        
+        [CLSCompliant(false)]
+        public uint GetUInt32() => uint.Parse(_value);
+        
+        [CLSCompliant(false)]
+        public ulong GetUInt64() => ulong.Parse(_value);
+
+        public bool TryGetByte(out byte value) => byte.TryParse(_value, out value);
+
+        public bool TryGetInt16(out short value) => short.TryParse(_value, out value);
+
+        public bool TryGetInt32(out int value) => int.TryParse(_value, out value);
+
+        public bool TryGetInt64(out long value) => long.TryParse(_value, out value);
+
+        public bool TryGetSingle(out float value) => float.TryParse(_value, out value);
+
+        public bool TryGetDouble(out double value) => double.TryParse(_value, out value);
 
-        public static implicit operator JsonNumber(byte value) { throw null; }
-        public static implicit operator JsonNumber(int value) { throw null; }
-        public static implicit operator JsonNumber(short value) { throw null; }
-        public static implicit operator JsonNumber(long value) { throw null; }
-        public static implicit operator JsonNumber(float value) { throw null; }
-        public static implicit operator JsonNumber(double value) { throw null; }
         [CLSCompliant(false)]
-        public static implicit operator JsonNumber(sbyte value) { throw null; }
+        public bool TryGetSByte(out sbyte value) => sbyte.TryParse(_value, out value);
+
         [CLSCompliant(false)]
-        public static implicit operator JsonNumber(ushort value) { throw null; }
+        public bool TryGetUInt16(out ushort value) => ushort.TryParse(_value, out value);
+
         [CLSCompliant(false)]
-        public static implicit operator JsonNumber(uint value) { throw null; }
+        public bool TryGetUInt32(out uint value) => uint.TryParse(_value, out value);
+
         [CLSCompliant(false)]
-        public static implicit operator JsonNumber(ulong value) { throw null; }
+        public bool TryGetUInt64(out ulong value) => ulong.TryParse(_value, out value);
+
+
+        public void SetFormattedValue(string value)
+        {
+            if (value == null)
+                throw new NullReferenceException("Expected number, but instead got null.");
+            if (value == "")
+                throw new ArgumentException("Expected number, but instead got empty string.");
+            
+            JsonWriterHelper.ValidateNumber(Encoding.UTF8.GetBytes(value).AsSpan());
+            _value = value;
```

Comment:
I would suspect these strings to generally be small (say < 256 characters). Consider instead, stackalloc a span and use the `Span<byte>` overload of `Encoding.UTF8.GetBytes` to avoid the allocation on .NET Core 3.0 (and use the arraypool for larger length strings).
https://github.com/dotnet/corefx/blob/ef5b086fc5f134e3244d0bca10d70db8055fa1c6/src/System.Runtime/ref/System.Runtime.cs#L7683

OR

Validate the value as chars.

Something to consider to optimize the implementation a bit (but that can be deferred to later, if you like).